sources/HUBViewModelDiff.h

```diff
@@ -24,6 +24,29 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+@class HUBViewModelDiff;
+
+typedef HUBViewModelDiff *(HUBDiffAlgorithm)(id<HUBViewModel>, id<HUBViewModel>);
```

Comment:
How come this is implemented as (_long_) C functions? Did you consider making this into a protocol instead? Such as:
```objc
@protocol HUBDiffAlgorithm
- (HUBViewModelDiff *)differenceGoingFromViewModel:(id<HUBViewModel>)fromViewModel
                                       toViewModel:(id<HUBViewModel>)toViewModel;
@end
```

It would, imo, allow for more flexibility. For example a `debugName` property could be added allowing the `-debugDescription` method further down to print which algorithm is used. It might also make the code feel less scary for non-C developers.

Not opposed to using passing function pointers like this. Just interested in your reasoning @Goos.